packages/framework/src/Model/Product/Search/Export/ProductSearchExportStructureFacade.php

```diff
@@ -45,11 +50,15 @@ public function __construct(Domain $domain, ElasticsearchStructureManager $elast
      */
     public function setElasticsearchStructureUpdateChecker(ElasticsearchStructureUpdateChecker $elasticsearchStructureUpdateChecker): void
     {
-        if ($this->elasticsearchStructureUpdateChecker !== null) {
+        if ($this->elasticsearchStructureUpdateChecker !== null && $this->elasticsearchStructureUpdateChecker !== $elasticsearchStructureUpdateChecker) {
             throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
         }
 
-        $this->elasticsearchStructureUpdateChecker = $elasticsearchStructureUpdateChecker;
+        if ($this->elasticsearchStructureUpdateChecker === null) {
+            @trigger_error(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
The message should be `sprintf('The %s() method is deprecated and will be removed in the next major. Use the constructor injection instead.', __METHOD__);`, you just copied the `BadMethodCallException` message.